ktor-io/common/src/io/ktor/utils/io/core/BufferFactory.kt

```diff
@@ -51,34 +51,33 @@ internal inline fun <R> withChunkBuffer(pool: ObjectPool<ChunkBuffer>, block: Ch
 }
 
 @ThreadLocal
-@Suppress("DEPRECATION")
-internal val DefaultChunkedBufferPool: ObjectPool<IoBuffer> = DefaultBufferPool()
+internal val DefaultChunkedBufferPool: ObjectPool<ChunkBuffer> = DefaultBufferPool()
 
 @Suppress("DEPRECATION")
 internal class DefaultBufferPool(
     private val bufferSize: Int = DEFAULT_BUFFER_SIZE,
     capacity: Int = 1000,
     private val allocator: Allocator = DefaultAllocator
-) : DefaultPool<IoBuffer>(capacity) {
+) : DefaultPool<ChunkBuffer>(capacity) {
 
-    override fun produceInstance(): IoBuffer {
-        return IoBuffer(allocator.alloc(bufferSize), null, this)
+    override fun produceInstance(): ChunkBuffer {
+        return ChunkBuffer(allocator.alloc(bufferSize), null, this)
     }
 
-    override fun disposeInstance(instance: IoBuffer) {
+    override fun disposeInstance(instance: ChunkBuffer) {
         allocator.free(instance.memory)
         super.disposeInstance(instance)
         instance.unlink()
     }
 
-    override fun validateInstance(instance: IoBuffer) {
+    override fun validateInstance(instance: ChunkBuffer) {
         super.validateInstance(instance)
 
-        if (instance === IoBuffer.Empty) {
-            error("IoBuffer.Empty couldn't be recycled")
+        if (instance === ChunkBuffer.Empty) {
```

Comment:
triple check for ChunkBuffer.Empty equality :)

Reply:
Could you tell me why it can be a problem?

Reply:
Thanks, will fix